types/lucene/index.d.ts

```diff
@@ -10,20 +10,37 @@ export interface TermLocation {
     offset: number;
 }
 
-export interface Node {
-    boost: null | number;
+export interface NodeField {
     field: string | '<implicit>';
-    fieldLocation: null;
+    fieldLocation: null | {
+        end: TermLocation;
+        start: TermLocation;
+    };
+}
+
+export type NodeTerm = NodeField & {
```

Comment:
Any reason to use a `type` and not an `interface` here (i.e. `export interface NodeTerm extends NodeField { ... }`).

Using interfaces allows users of the library to augment the types in their projects, if they need to - while types don't.